### PR TITLE
H-100: Add scaffolding for `@local/hash-graph-types`

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

This adds the basic scaffolding for the implementation of schema validation, where the majority of the logic of `hash-ai-worker-py` will live.

## 🔍 What does this change?

- Add basic setup for the Python library
- Add a dependency for `hash-ai-worker-py`
- Rework some script invocations to make it work with internal dependencies

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

 - [x] is internal and does not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] affected the execution graph, and the `turbo.json`'s have been updated to reflect this

### CODE REVIEW

File: libs/@local/hash-graph-types/LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# License
```
Comment (nonparibus) [inline libs/@local/hash-graph-types/LICENSE.md:1]: @TimDiekmann Shouldn't this be AGPL?

File: libs/@local/hash-graph-types/pyproject.toml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 name = "hash-graph-types"
 version = "0.0.0"
 description = "Type definitions for the type system and entity validation based on schemas"
-license = "AGPL-3.0-only"
+license = "MIT OR Apache-2.0"
```
Comment (nonparibus) [inline libs/@local/hash-graph-types/pyproject.toml:5]: Why is this not AGPL like the HASH application?

File: libs/@local/hash-graph-types/LICENSE.md (hunk 1 above)
Comment (TimDiekmann) [reply libs/@local/hash-graph-types/LICENSE.md:1]: I was assuming as we want it at the awesome-blocks page and this is only a small library that "only" transforms some JSON schemas we were about to publicize it under MIT/Apache, especially as the [type-system is also MIT/Apache](https://github.com/blockprotocol/blockprotocol/blob/main/libs/%40blockprotocol/type-system/crate/LICENSE.md). Still, I was about to check with you as soon as the PR is ready for review.

File: libs/@local/hash-graph-types/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+description = "Type definitions for the type system and entity validation based on schemas"
+license = "MIT OR Apache-2.0"
+authors = ["HASH"]
+readme = "README.md"
+packages = [{ include = "graph_types" }]
+
+[tool.poetry.dependencies]
+python = "^3.11"
+
+pydantic = "^2.0"
```
Comment (TimDiekmann) [inline libs/@local/hash-graph-types/pyproject.toml:13]: Added in this PR to add the `pydantic.mypy` plugin

File: libs/@local/hash-graph-types/LICENSE.md (hunk 1 above)
Comment (TimDiekmann) [reply libs/@local/hash-graph-types/LICENSE.md:1]: You want me to change it to AGPL?

Comment (nonparibus) [reply libs/@local/hash-graph-types/LICENSE.md:1]: Let's make this AGPL for now, as it's only at the point of being `@local` anyway (even if we are planning to link from `awesome-blocks`). We can can examine this in more depth with a view to liberalizing it later. Thanks!

File: libs/@local/hash-graph-types/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# HASH Graph types
+
+Type definitions for the type system and entity validation based on schemas.
```
Comment (nonparibus) [inline libs/@local/hash-graph-types/README.md:3]: I wonder if we can make this a bit more descriptive. @CiaranMn for thoughts.

File: libs/@local/hash-graph-types/graph_types/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+"""Replicates the Block Protocol type system to be used in Python."""
```
Comment (nonparibus) [inline libs/@local/hash-graph-types/graph_types/__init__.py:1]: "to be used" -> "for use"?

```suggestion
"""Replicates the Block Protocol type system for use in Python."""
```

**Review by nonparibus — APPROVED** [review]
Legal ✅ but a couple of comments (CCd @CiaranMn)

File: libs/@local/hash-graph-types/README.md (hunk 4 above)
Comment (nonparibus) [reply libs/@local/hash-graph-types/README.md:3]: We should also update the package metadata to reflect the revised description here when added.

**Review by indietyp — DISMISSED** [review]
LGTM, thank you! 🎉

**Review by indietyp — APPROVED** [review]

**Review by indietyp — APPROVED** [review]

**Comment by indietyp** [comment]
The build simplifications look great btw!

**Review by nonparibus — APPROVED** [review]

File: libs/@local/hash-graph-types/README.md (hunk 4 above)
Comment (TimDiekmann) [reply libs/@local/hash-graph-types/README.md:3]: I propose to do this in a follow-up so we can get this PR out of the way.

**Comment by TimDiekmann** [comment]
> The build simplifications look great btw!

I would have preferred to do this in a separate PR but it was kind of required. Thanks, though!

**Review by nonparibus — APPROVED** [review]
Thanks, @TimDiekmann!

**Pull request merged**